### PR TITLE
feat(swc-angular-plugin): add support for latest @swc/core

### PR DESCRIPTION
**Pull request by pumano**

### CODE REVIEW

File: packages/swc-angular-plugin/src/component_decorator_visitor.rs [hunk 1]
```diff
@@ -96,10 +98,11 @@ impl VisitMut for ComponentDecoratorVisitor {
 
         if key.sym.eq("styleUrls") {
             node.key = PropName::Ident(Ident {
+                ctxt: SyntaxContext::default(),
```
Comment (pumano) [inline packages/swc-angular-plugin/src/component_decorator_visitor.rs:101]: i'm not sure what should be used here, but `default` works. Previous version of Ident does not have that required field.

File: packages/swc-angular-plugin/src/testing.rs [hunk 2]
```diff
@@ -12,10 +12,10 @@ use swc_ecma_parser::{Syntax, TsConfig};
 /// The typical example is when we add a decorator call in the middle of a function
 /// with no indentation.
 pub fn test_visitor<V>(visitor: V, input: &str, expected: &str) where V: VisitMut {
-    let syntax = Syntax::Typescript(TsConfig::default());
+    let syntax = Syntax::Typescript(TsSyntax::default());
     let transform = as_folder(visitor);
     let actual = Tester::run(|tester| {
-        let result = tester.apply_transform(transform, "input.js", syntax, input)?;
+        let result = tester.apply_transform(transform, "input.js", syntax, Some(true), input)?;
```
Comment (pumano) [inline packages/swc-angular-plugin/src/testing.rs:18]: added new parameter `"is_module"` means is esm module, all typescript components is esm modules when use `import` and `export`

File: packages/swc-angular/package.json [hunk 3]
```diff
@@ -12,6 +12,6 @@
   ],
   "peerDependencies": {
     "@jscutlery/swc-angular-plugin": ">=0.11.0 <1.0.0",
-    "@swc/core": ">=1.4.0 <1.7.0"
+    "@swc/core": ">=1.7.0 <1.10.0"
```
Comment (pumano) [inline packages/swc-angular/package.json:15]: version should be bumped due to breaking changes (@swc/core 1.7+ supported only now)

**Comment by pumano** [comment]
@yjaaidi could you please check it?

**Review by edbzn — APPROVED** [review]

File: packages/swc-angular-plugin/src/component_decorator_visitor.rs (hunk 1 above)
Comment (yjaaidi) [reply packages/swc-angular-plugin/src/component_decorator_visitor.rs:101]: weird but default lgtm

File: packages/swc-angular-plugin/src/component_property_visitor/component_property_visitor_test.rs [hunk 4]
```diff
@@ -26,6 +26,34 @@ fn test_input() {
     );
 }
 
+#[test]
```
Comment (yjaaidi) [inline packages/swc-angular-plugin/src/component_property_visitor/component_property_visitor_test.rs:29]: just curious, what is the problem that this test detects?
was there a case where services affected the inputs transform?

File: tests/swc-angular-wide/src/swc-plugin-angular-input.spec.ts [hunk 5]
```diff
@@ -27,6 +27,38 @@ describe('swc-angular-plugin: input', () => {
     expect(heading).toBe('Hello input!');
   });
 
+  it('should bind input data via setInput', () => {
```
Comment (yjaaidi) [inline tests/swc-angular-wide/src/swc-plugin-angular-input.spec.ts:30]: same here, what is the purpose of this test?

**Review by yjaaidi — CHANGES_REQUESTED** [review]
Thank you @pumano for the contribution! Very much appreciated.
Before merging this, could you please remove the tests that you've added as I don't see the value they provide compared to existing tests. But, I missed something about their value, please let me know.

File: packages/swc-angular-plugin/src/component_property_visitor/component_property_visitor_test.rs (hunk 4 above)
Comment (pumano) [reply packages/swc-angular-plugin/src/component_property_visitor/component_property_visitor_test.rs:29]: that I try to test problem where #304 described, but looks like problem not here, I can delete that tests

**Comment by yjaaidi** [comment]
👌

**Pull request merged**